src/libraries/System.Data.Common/src/System/Data/Common/ObjectStorage.cs

```diff
@@ -341,7 +341,7 @@ public override object ConvertXmlToObject(string s)
 
             StringReader strreader = new StringReader(s);
             XmlSerializer deserializerWithOutRootAttribute = ObjectStorage.GetXmlSerializer(type);
-            return (deserializerWithOutRootAttribute.Deserialize(strreader));
+            return (deserializerWithOutRootAttribute.Deserialize(strreader))!;
```

Comment:
wondering if this should return nullable instead but perhaps might be better to file an issue rather than solve here